### PR TITLE
Check index_dtype and data_dtypes more strictly.

### PR DESCRIPTION
**Pull request by ueshin**
Checks explicitly `index_dtype` and `data_dtypes` are the instance of `np.dtype` or `ExtensionDtype`.

### CODE REVIEW

File: databricks/koalas/internal.py [hunk 1]
```diff
@@ -593,13 +595,15 @@ def __init__(
...
             spark_type_to_pandas_dtype(spark_frame.select(scol).schema[0].dataType)
-            if dtype is None
+            if dtype is None or dtype == np.dtype("object")
             else dtype
             for dtype, scol in zip(data_dtypes, data_spark_columns)
         ]
 
         assert all(
-            as_spark_type(dtype, raise_error=False) is not None for dtype in data_dtypes
+            isinstance(dtype, Dtype.__args__)  # type: ignore
```
Comment (xinrong-databricks) [inline databricks/koalas/internal.py:604]: Can I take it as pandas dtype consists of np.dtype and ExtensionDtype?

**Review by xinrong-databricks — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
Looks great, thank you!

File: databricks/koalas/internal.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/internal.py:604]: I believe so, so far. https://github.com/pandas-dev/pandas/blob/7af47c9/pandas/_typing.py#L126

Comment (xinrong-databricks) [reply databricks/koalas/internal.py:604]: Got it, thanks!

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**